core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2334,3 +2335,26 @@ strip eq arr@Array{..} = do
             then getLast prev low
             else return cur
         else return cur
+
+-- | Bubble sorts the elements.
+--
+-- /Pre-release/
+{-# INLINE bubbleAsc #-}
+bubbleAsc :: (MonadIO m, Storable a) =>
+    Int -> (a -> a -> Ordering) -> Array a -> m ()
+bubbleAsc idx cmp0 arr =
+    do
+    let l = length arr
+    go idx l
```

Comment:
Why `l` is being passed to `go` if it never changes? It can be defined as a `where` binding.